src/charm.py

```diff
@@ -401,17 +415,13 @@ def _unit_ip(self) -> str:
         return str(self.model.get_binding(Config.Relations.PEERS).network.bind_address)
 
     @property
-    def is_external_client(self) -> Optional[str]:
+    def is_external_client(self) -> bool:
         """Returns the connectivity mode which mongos should use.
 
-        This is determined by checking the modes requested by the client(s).
-
-        TODO: Future PR. This should be modified to work for many clients.
+        Note that for K8s routers this should always default to True. However we still include
```

Comment:
It is not true. Yes, K8s mongos should always expose the port (instead of the Unix socket on VM).
BUT mongos can be used outside of Kubernetes and can be used inside the Kubernetes.
We shouldn't open public ports for setups that don't need connectivity outside of Kubernetes. This is a major security issue.
Such configurations should be controlled by config option, e.g. https://github.com/canonical/kafka-k8s-operator/blob/main/config.yaml#L111-L114

Reply:
@delgod Okay thank you for this message. This aligns with what I had initially thought.

I got confused [by an earlier conversation](https://github.com/canonical/mongos-k8s-operator/pull/19#discussion_r1714006407) .  That referred to a quote from [spec DA073](https://docs.google.com/document/d/1Y7OZWwMdvF8eEMuVKrqEfuFV3JOjpqLHL7_GPqJpRHU/edit?usp=sharing):

> The data-integrator should raise the flag by default. Regular database charms and K8s routers can ignore the flag, as it does not require them to change behaviour.

Perhaps spec needs updated?